plugin/pkg/admission/serviceaccount/admission.go

```diff
@@ -292,10 +297,21 @@ func (s *serviceAccount) limitSecretReferences(serviceAccount *api.ServiceAccoun
 			continue
 		}
 		secretName := source.Secret.SecretName
-		if !referencedSecrets.Has(secretName) {
+		if !mountableSecrets.Has(secretName) {
```

Comment:
Service accounts are a likely way to gate access to secrets eventually. The admission controller has a "limitSecrets" option (false by default for now) that only lets the pod use secrets that are referenced by its service account.